### PR TITLE
fix(router): throw an error when navigate to null/undefined path

### LINKED ISSUE #10560: Router.navigate fails with unhelpful error if `undefined` is part of the path
**I'm submitting a ...**

```
[ ] bug report
[X] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

An unhelpful error is thrown when `undefined` appears in a `Router.navigate([…])` path. For example, here's what happens when I call `this.router.navigate(['/foo/, undefined])`:

example-stack-trace

The line number referenced - `SessionComponent - inline template 26` - is `<button (click)="nextStep()" />`.

**Expected/desired behavior**

It would be fantastic if the error was something along the lines of `the requested path - ['/foo', undefined] - contains an undefined component`.

**Please tell us about your environment:**
- **Angular version:** 2.0.0-rc.4

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes https://github.com/angular/angular/issues/10560

### CODE REVIEW

File: modules/@angular/router/src/router.ts [hunk 1]
```diff
@@ -558,6 +559,15 @@ export class Router {
     }
   }
 
+  private validateCommands(commands: string[]): void {
+    for (let i = 0; i < commands.length; i++) {
```
Comment (vicb) [inline modules/@angular/router/src/router.ts]:
```ts
commands.forEach(cmd, i => ...
   `... contains ${cmd} at index ${i}...`
```

File: modules/@angular/router/src/router.ts [hunk 2]
```diff
@@ -558,6 +559,15 @@ export class Router {
     }
   }
 
+  private validateCommands(commands: string[]): void {
```
Comment (vicb) [inline modules/@angular/router/src/router.ts]: make a function outside of the class. Thanks

File: modules/@angular/router/src/router.ts (hunk 1 above)
Comment (DzmitryShylovich) [inline modules/@angular/router/src/router.ts]: `forEach` skips undefined elements

Comment (vicb) [inline modules/@angular/router/src/router.ts]: Ah right it would skip holes... (but not undefined els) I guess a `for` is better then.

**Comment by matsko** [comment]
The v4 commit is https://github.com/angular/angular/commit/46cb04d575ba5ec6ac50719667146ab3e2c52bdc

**Pull request closed without merging**